### PR TITLE
move userprofile to settings, add views for workinghours

### LINKED ISSUE #281: Overview for WorkingHours
As a planner, I'd like to have a view where I can see a ranking of volunteers by their worked hours (events and workinghours)

### LINKED ISSUE #607: Manage working hours for other users
Delete, add and edit working hours for the users you have workhours management permissions for.

Also, don't show the request button on a user profile if you have create-permissions. The CRUD views should be used instead.

### LINKED ISSUE #770: Refactor and split up UserProfileView
- Persönliche Informationen werden Einstellungsseite (dann erstmal readonly)
    -  inkl. Gruppenmitgliedschaften und Qualifikationen anzeigen
- Kalender --> eigene Einstellungsseite
- Eigene Views für Arbeitsstunden
    - Statistiken, Management/CRUD (see also #607)
    - ist keine Einstellung, also evtl. für Nutzer im Dropdown, Verwalter kriegen vlt. extra “Statistik” main entry?!

### PR DESCRIPTION
**Pull request by jeriox**
closes #770 closes #607 closes #281

- [x] edit and delete views for workinghours (see #607)?
- [x] respect granting permissions on group level
- [x] allow timeframe to be selected (mininmal version: select year)
- [x] make layout of workinghourlist better (no card)
- [ ] display nice statistics on overview page -> moved to #837 
- [ ] export button for excel sheet (via API)? -> moved to #838

### CODE REVIEW

File: ephios/core/urls.py [hunk 1]
```diff
@@ -210,21 +218,11 @@
         UserProfileCreateView.as_view(),
         name="userprofile_create",
     ),
-    path(
-        "users/<int:pk>/",
-        UserProfileDetailView.as_view(),
-        name="userprofile_detail",
```
Comment (felixrindt) [inline ephios/core/urls.py:216]: The user list at `/users/` still tries to reverse this view.

File: ephios/core/forms/users.py [hunk 2]
```diff
@@ -284,29 +281,35 @@ def __init__(self, *args, **kwargs):
...
-            Field("when"),
+            Field("date"),
             Field("hours"),
             Field("reason"),
             FormActions(
                 Submit("submit", _("Send"), css_class="float-end"),
-                AbortLink(href=reverse("core:profile")),
+                AbortLink(href=reverse("core:workinghour_detail", args=[self.user.pk]))
+                if self.can_grant
+                else AbortLink(href=reverse("core:workinghour_own")),
```
Comment (felixrindt) [inline ephios/core/forms/users.py:305]: I must say I find having the Abort Links build inside the form kind of defies the django pattern of having forms reusable. Like that kind of navigation should be part of the view, right?

**Comment by jeriox** [comment]
Respecting the permissions on the group level still feels a bit hacky as it always requires two separate queries to check if the target user is in _any_ of the groups that the current user is allowed to grant workinghours for.

File: ephios/core/views/workinghour.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+from django.utils.translation import gettext as _
+from django.views.generic import DeleteView, DetailView, FormView, TemplateView, UpdateView
+from guardian.shortcuts import get_objects_for_user
+
+from ephios.core.forms.users import WorkingHourRequestForm
+from ephios.core.models import LocalParticipation, UserProfile, WorkingHours
+from ephios.extra.mixins import CustomPermissionRequiredMixin
+
+
+class WorkingHourPermissionMixin:
```
Comment (jeriox) [inline ephios/core/views/workinghour.py:21]: @felixrindt I'm not super happy with this mixin, as there is a lot of similar checking across all views, but not always exactly the same. Maybe you can come up with something that is a bit cleaner?

File: ephios/core/models/users.py [hunk 4]
```diff
@@ -154,20 +155,22 @@ def get_workhour_items(self):
         from ephios.core.models import AbstractParticipation
 
         participations = (
-            self.participations.filter(state=AbstractParticipation.States.CONFIRMED)
+            self.participations.filter(state=AbstractParticipation.States.CONFIRMED, finished=True)
```
Comment (felixrindt) [inline ephios/core/models/users.py:158]: I think this is not the right filter. Should be events that ended, while `finished` means ephios has processed triggers for this shift being finished.

File: ephios/core/templatetags/user_extras.py [hunk 5]
```diff
@@ -74,3 +74,8 @@ def shifts_needing_disposition(user):
         )
         .filter(request_count__gt=0)
     )
+
+
+@register.filter(name="allows_grant_for")
+def allows_grant_for(grant_group_ids, user_group_ids):
+    return bool(grant_group_ids & user_group_ids)
```
Comment (felixrindt) [inline ephios/core/templatetags/user_extras.py:81]: Now I see why you picked the name but just looking at this line something like `intersects` might be the better name for the filter

**Review by felixrindt — APPROVED** [review]

**Pull request merged**